internal/net/grpc/pool.go

```diff
@@ -45,24 +67,104 @@ type ClientConnPool struct {
 	closing atomic.Value
 }
 
-const (
-	localIPv4   = "127.0.0.1"
-	localHost   = "localhost"
-	defaultPort = "80"
-)
+func NewPool(ctx context.Context, addr string, size uint64, dopts ...DialOption) (ClientConnPool, error) {
+	var ok bool
+	host, port, isIP, err := net.Parse(addr)
+	if err != nil {
+		log.Warn(err)
+		if len(host) == 0 {
+			host = addr
+		}
+		p, ok := scanGRPCPort(ctx, host)
+		if ok {
+			port = p
+		}
+	}
+	cp, err := newPool(ctx, host, port, size, dopts...)
+	if err != nil {
+		if cp != nil {
+			err = errors.Wrap(cp.disconnect(), err.Error())
+		}
+		port, ok = scanGRPCPort(ctx, host)
+		if !ok {
+			return nil, err
+		}
+		cp, err = newPool(ctx, host, port, size, dopts...)
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg?style=flat-square)
the `err` should be handled.

Reply:
I forgot about that I'll fix them soon

Reply:
fixed